File: java-framework/src/main/java/webui/automation/elements/ContainerElement.java
package webui.automation.elements;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.SearchContext;
import org.openqa.selenium.WebElement;

import webui.automation.framework.BaseElement;

public class ContainerElement extends BaseElement implements SearchContext {

    public ContainerElement(By locator) {
        super(locator);
    }

    /**
     * @param  webElement  The {@link WebElement} of this container element.
     * @throws NullPointerException if the specified <code>webElement</code> is <code>null</code>
     */
    public ContainerElement(WebElement webElement) {
        super(webElement);
    }

    ////// Implements the SearchContext interface.  //////
    
    public WebElement findElement(By by) {
        return getWebElement().findElement(by);
    }

    public List<WebElement> findElements(By by) {
        return getWebElement().findElements(by);
    }

}

File: cs-framework/WebUI-Automation-Framework/Elements/ContainerElement.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;  // ReadOnlyCollection<T>
using OpenQA.Selenium;
using WebUI.Automation.Framework;

namespace WebUI.Automation.Elements
{

    public class ContainerElement : BaseElement, ISearchContext
    {

        public ContainerElement(By locator) : base(locator)
        {
        }

        /// <param name="webElement">  The <seealso cref="WebElement"/> of this container element. </param>
        /// <exception cref="NullPointerException"> if the specified <code>webElement</code> is <code>null</code> </exception>
        public ContainerElement(IWebElement webElement) : base(webElement)
        {
        }

        ////// Implements the SearchContext interface.  //////

        public virtual IWebElement FindElement(By by)
        {
            return WebElement.FindElement(by);
        }

        public virtual ReadOnlyCollection<IWebElement> FindElements(By by)
        {
            return WebElement.FindElements(by);
        }

    }

}